configs/seL4Config.cmake

```
@@ -227,11 +227,18 @@ if(KernelArchARM)
     config_set(KernelArmMach ARM_MACH "${KernelArmMach}")
 endif()
 
-if("${TRIPLE}" STREQUAL "")
+set(LLVM_TOOLCHAIN ON CACHE BOOL "if using clang as compiler (auto detected)")
+if (CMAKE_ASM_COMPILER_ID MATCHES "GNU")
+    set(LLVM_TOOLCHAIN OFF)
+endif()
+
+
+if("$LLVM_TOOLCHAIN" STREQUAL "ON")
```

nitpicking:
```suggestion
if("${LLVM_TOOLCHAIN}" STREQUAL "ON")
```
We could also simply use `if(LLVM_TOOLCHAIN)`, as this is sufficient given the if-evaulation rules from https://cmake.org/cmake/help/latest/command/if.html#command:if